app/controllers/fhir/r4/api_controller.rb

```diff
@@ -266,6 +275,17 @@ def transaction
     contents.entry&.each_with_index do |entry, index|
       next unless entry.resource&.resourceType&.downcase == 'patient'
 
+      # Duplicate detection check
+      if entry&.request&.ifNoneExist.present?
+        matches = search_patients(Rack::Utils.parse_nested_query(entry&.request&.ifNoneExist))
+        num_matches = matches.size
+        status_ok(matches.first.as_fhir) && return if num_matches == 1
```

Comment:
Here I don't think we can just return, since we haven't necessarily processed the entire transaction. I think what we want to do is proceed, not create a new patient, and instead of having 201 created in the transaction response:
![image](https://user-images.githubusercontent.com/52747882/136616996-e88c0373-fae6-4c8e-998e-71e8b66aa656.png)

We will want to have `200 OK`. Then per @Bialogs's suggestion, according to the spec we do not actually want to return the Patient resource, just the `200 OK`, so `Bundle.entry.resource` will actually be empty for that element.

Reply:
Good point. That's an interesting question. As of right now, we don't support creating Observations that point to Patients which are not within the same Bundle. So an Observation cannot point to an existing Patient. If it does, we fail with a 422 error. So it feels like if the Patient already exists, it should be a 422 error, and the entire transaction should be canceled.

Reply:
Yes exactly, assuming the `Bundle.entry[1]` is the lab pointing to the duplicate patient.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/dfabb5dca271e1d1f47b09680a1dd63cd63f6218#diff-402d451d1f7019322100522f6898c3153444edb8ddb963e181a457c84a2ca6b6R1244